modules/unsupported/flatgeobuf/src/test/java/org/geotools/data/flatgeobuf/FlatGeobufDataStoreTest.java

```diff
@@ -903,6 +903,24 @@ public void readCountriesFids() throws IOException {
         }
     }
 
+    @Test
+    public void readCountriesOffsetLimit() throws IOException {
+        SimpleFeatureSource featureSource = getFeatureSource("countries");
+        SimpleFeatureType schema = featureSource.getSchema();
+        Query query = new Query(schema.getTypeName());
+        query.setMaxFeatures(2);
+        query.setStartIndex(10);
+        SimpleFeatureCollection featureCollection = featureSource.getFeatures(query);
+        try (SimpleFeatureIterator it = featureCollection.features()) {
+            SimpleFeature f1 = it.next();
+            assertEquals("countries.107", f1.getID());
+            SimpleFeature f2 = it.next();
+            assertEquals("countries.108", f2.getID());
+            boolean hasNext = it.hasNext();
+            assertFalse(hasNext);
```

Comment:
Against a database one would need to use a order by to get a predictable output sequence, but it's very easy to imagine a file read to have a predictable order instead. Just trying to make sure.

Reply:
Hmm yes, and FlatGeobuf doesn't allow for any mutation so ordering can be assumed to be static.